src/charm.py

```diff
@@ -78,6 +84,19 @@ def unit_peer_data(self) -> MutableMapping[str, str]:
 
         return self.peer_relation.data[self.unit]
 
+    def _on_storage_attached(self, event: StorageAttachedEvent) -> None:
+        path = event.storage.location if event.storage else None
+        if not path:
+            logger.error("Unable to find storage in StorageAttachedEvent")
+            event.defer()
+            return
+
+        self.unit_peer_data.update({"logs": "attached"})
+
+    def _on_storage_detatched(self, _: StorageDetachingEvent) -> None:
```

Comment:
nit: `detached`, also is this callback registered?

Reply:
Valid nit.
It's not, it'll be in future PRs. This isn't merging to `main`, just to a feature branch.

Reply:
It's left out because I need to actually validate the behavior of Kafka when you rug-pull a directory from it.